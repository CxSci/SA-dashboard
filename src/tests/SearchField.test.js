/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import SASearchField from "../components/SASearchField";
import { preRender } from "./testUtils";
import Enzyme from 'enzyme';
import { resetState } from "../redux/action-creators";
import { store } from "../redux/store";
import { loadFeatures } from "../redux/reducer";
import {loadGeoJSON} from './testUtils';

beforeEach(() => {
  resetState()
  fetchMock.mockIf("SA_dashboard.geojson", loadGeoJSON("./src/data/SA_dashboard.geojson"))
  store.dispatch(loadFeatures())
})

it("renders correctly", async () => {
  const tree = renderer.create(preRender(<SASearchField />, store)).toJSON();
  expect(tree).toMatchSnapshot();
});

it("render dropdown on search input", async () => {
  const event = {target: {value: "ad"}};
  const component = Enzyme.mount(preRender(<SASearchField />, store));
  component.find('input').simulate('mouseEnter').simulate('change', event)
  expect(component.debug()).toMatchSnapshot();

})

it("can select feature", () => {
  const event = {target: {value: "ad"}};
  const component = Enzyme.mount(preRender(<SASearchField />, store));
  component.find('input').simulate('mouseEnter').simulate('change', event)
  console.log(component.debug())
  component.find("#downshift-3-item-0").simulate("click")
  //console.log(component.debug())

})