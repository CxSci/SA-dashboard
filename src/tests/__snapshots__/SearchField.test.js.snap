// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sidebarHeader "
>
  <div
    className="bigTitle"
  >
    <div
      className="titleButton"
      onClick={[Function]}
    >
      <div
        className="titleUnderline"
      />
    </div>
    <div
      className="cancelButton"
      onClick={[Function]}
    >
      <div
        className="cancelUnderline"
      >
        <Mock SVG />
      </div>
    </div>
  </div>
  <div
    className="searchContainer"
  >
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-owns="downshift-0-menu"
      className="searchField"
      role="combobox"
    >
      <input
        aria-autocomplete="list"
        aria-controls="downshift-0-menu"
        aria-labelledby="downshift-0-label"
        autoComplete="off"
        id="downshift-0-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search by suburb or region"
        spellCheck="false"
        value=""
      />
      <button
        aria-label="toggle menu"
        id="downshift-0-toggle-button"
        onClick={[Function]}
        tabIndex={-1}
      >
        <Mock SVG />
      </button>
      <div
        style={
          Object {
            "left": "0",
            "position": "absolute",
            "top": "0",
            "zIndex": 1,
          }
        }
      >
        <ul
          aria-labelledby="downshift-0-label"
          className="searchMenu"
          hidden={true}
          id="downshift-0-menu"
          onMouseLeave={[Function]}
          role="listbox"
        />
      </div>
    </div>
  </div>
</div>
`;
